### PR TITLE
HTBHF-1701 Updated 'Request a new code' link to redirect to send code…

### PR DESCRIPTION
**Pull request by dwybourn**
Request a new code link now takes the user back to send code page.

### CODE REVIEW

File: src/test/common/steps/enter-code-steps.js [hunk 1]
```diff
@@ -23,3 +23,9 @@ Then(/^I am informed that I must enter in the code that was sent to me$/, async
     pages.enterCode.getConfirmationCodeLinkErrorCss(),
     'Enter the 6 digit code we sent you')
 })
+
+Then(/^The request a new code link points to the send code page$/, async function () {
+  const requestNewCodeLink = await pages.enterCode.getRequestNewCodeLink()
+  const requestNewCodeLinkHref = await requestNewCodeLink.getAttribute('href')
+  expect(requestNewCodeLinkHref).to.be.equal(`${pages.url}/send-code`)
```
Comment (mikeSmith08) [inline src/test/common/steps/enter-code-steps.js]: You should be getting the `/send-code` link from the SendCode page object.

File: src/web/views/enter-code.njk [hunk 2]
```diff
@@ -33,6 +33,6 @@
     {% endcall %}
 
     {# TODO DW HTBHF-1712 update link to send new confirmationCode  #}
-    <p class="govuk-body-m"><a href="#">{{ requestANewCode }}</a> {{ ifItDoesNotArrive }}</p>
+    <p class="govuk-body-m"><a href="/send-code">{{ requestANewCode }}</a> {{ ifItDoesNotArrive }}</p>
```
Comment (mikeSmith08) [inline src/web/views/enter-code.njk]: Can we add an id to this link so we can use `findById` in `Page`? We then don't need the new method `findByLinkText`.

File: src/test/common/page/page.js [hunk 3]
```diff
@@ -124,6 +124,15 @@ class Page {
     }
   }
 
+  async findByLinkText (linkText) {
```
Comment (mikeSmith08) [inline src/test/common/page/page.js]: See my comment below about adding an id to the link, we then don't need this method and can use `findById` which will be less fragile should the text on the link change.

File: src/test/common/steps/enter-code-steps.js (hunk 1 above)
Comment (dwybourn) [reply src/test/common/steps/enter-code-steps.js]: Updated to use `pages.sendCode.getPath()`

File: src/web/views/enter-code.njk (hunk 2 above)
Comment (dwybourn) [reply src/web/views/enter-code.njk]: Added id.

File: src/test/common/page/page.js (hunk 3 above)
Comment (dwybourn) [reply src/test/common/page/page.js]: Removed method.

**Pull request merged**